### PR TITLE
Fix outstanding balance sum in payment report

### LINKED ISSUE #7333: Payment reports broken
## Description

Bugnag errors are coming up on payment reports this morning. They are almost certainly hitting the Snail of Doom.

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/606d6a7b990dac0008c9a04c?filters[event.since][0]=30d&filters[error.status][0]=open

```
TypeError spree/admin/reports#payments
OrderBalance can't be coerced into BigDecimal
```

## Expected Behavior

Payment reports load.

## Actual Behaviour

Fatal errors thrown.

## Steps to Reproduce

Load a payment report with a bunch of orders.

## Workaround

None.

## Severity

bug-s1: a critical feature is broken: checkout, payments, signup, login
bug-s2: a non-critical feature is broken, no workaround

## Possible Fix

We'll need to change line `lib/open_food_network/payments_report.rb:103`, and any other places where we use `Enumerable#sum` with order balances.

`OrderBalance` has a `+` method to work with `#sum`, but a comment in the source code for sum says:

```
Enumerable#sum method may not respect method redefinition of “+” methods such as Integer#+.
```
turns out this is true :warning:

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #7333. See notes in issue.

#### What should we test?

Payment reports should load without error

#### Release notes

Fixed outstanding balance sum in payment report

Changelog Category: User facing changes

### CODE REVIEW

**Comment by filipefurtad0** [comment]
Thanks so much for the quick fix on this @Matt-Yorkley :+1: 

The reports responsible for the two bugsnag errors (and snails) are rendering now, after this PR. I figured it might be useful to test this under two scenarios, in which customer balances feature is toggled **on** and **off**. Found no difference whatsoever in the outcome:

- under `/admin/reports/payments`

Itemized totals:

image

Payment totals:

image

Notice the strangely long decimal. Looks a bit weird - I'm guessing they were the root cause? I've seen this recently... (under products somehow...?)

Placed an order and checked that the customer balance is updated. The weird looking zeroes disappear if one selects "all" from the dropdown, and re-appear if a hub is selected:

Peek 2021-04-07 12-19

Peek 2021-04-07 12-18

I'm not sure this is related. Just felt it's worth pointing out since the Bugsnag does mention `BigDecimal`s


It's ready to go.

File: lib/open_food_network/payments_report.rb [hunk 1]
```diff
@@ -100,7 +100,7 @@ def columns
          proc { |orders| orders.first.distributor.name },
          proc { |orders| orders.to_a.sum(&:item_total) },
          proc { |orders| orders.sum(&:ship_total) },
-         proc { |orders| orders.sum(&:outstanding_balance) },
```
Comment (sauloperez) [inline lib/open_food_network/payments_report.rb:103]: I think we were just missing https://www.rubydoc.info/stdlib/bigdecimal/BigDecimal#coerce-instance_method. I didn't know this was possible :heart_eyes: 

I wonder if we didn't spot this due to the Ruby upgrade that after this :thinking:

Comment (Matt-Yorkley) [reply lib/open_food_network/payments_report.rb:103]: It's really weird. I tried 5 or 6 times to write a test for this, in feature specs, controller specs, model specs... it passed without error every time! And we actually _do_ have a test for this already, and it was passing. Really easy to replicate in the browser or console though! :boom:

#WTF :man_shrugging: :man_shrugging: :man_shrugging:

Comment (sauloperez) [reply lib/open_food_network/payments_report.rb:103]: :thinking: :thinking: I was probably stubbing things I shouldn't?

Comment (Matt-Yorkley) [reply lib/open_food_network/payments_report.rb:103]: I would have expected the tests here to have failed:

https://github.com/openfoodfoundation/openfoodnetwork/blob/7d5726b6c799cd0229640eaa503d6474e64bae3d/spec/features/admin/reports/payments_report_spec.rb

**Comment by filipefurtad0** [comment]
> Notice the strangely long decimal. Looks a bit weird - I'm guessing they were the root cause? I've seen this recently... (under products somehow...?)

Just met this again (for the record, `unit_prices` is **off**):

Peek 2021-04-21 21-38

This value gets actually gets stored and is even displayed in the shopfront:

image

I wonder if this can mess anything throughout the app...

**Comment by sauloperez** [comment]
This definitely needs an issue. We're missing some rounding somewhere.

**Comment by filipefurtad0** [comment]
Opened #7504.

**Pull request merged**